src/NLog.StructuredLogging.Json/NestedContext.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace NLog.StructuredLogging.Json
+{
+    internal class NestedContext : IDisposable, INestedContext
+    {
+        private readonly object _lock = new object();
+        private readonly ILogger _logger;
+        private readonly IDisposable _disposable;
+        private readonly NestedContext _parentScope;
+        private readonly Dictionary<string, object> _properties;
+        private CalculatedContext _calculatedLogProperties;
+
+        public Guid ScopeId { get; }
+        public string Scope { get; }
+        public string ScopeIdTrace { get; }
+        public string ScopeNameTrace { get; }
+        public IReadOnlyDictionary<string, object> Properties => _properties;
+
+
+        public NestedContext(ILogger logger, string scope, IDictionary<string, object> properties)
```

Comment:
Do we need to validate parameters here, or will `null`s etc. get handled elsewhere?